### PR TITLE
[GlobalISel][AArch64] Legalize G_INSERT_VECTOR_ELT for SVE

### PR DESCRIPTION
**Pull request by tschuett**
There are patterns for:
* {nxv2s32, s32, s64},
* {nxv4s16, s16, s64},
* {nxv2s16, s16, s64}

### CODE REVIEW

File: llvm/include/llvm/CodeGen/GlobalISel/LegalizerInfo.h [hunk 1]
```diff
@@ -273,6 +273,11 @@ inline LegalityPredicate typeIsNot(unsigned TypeIdx, LLT Type) {
 LegalityPredicate
 typePairInSet(unsigned TypeIdx0, unsigned TypeIdx1,
               std::initializer_list<std::pair<LLT, LLT>> TypesInit);
+/// True iff the given types for the given tuple of type indexes is one of the
+/// specified type tuple.
+LegalityPredicate
+typeTupleInSet(unsigned TypeIdx0, unsigned TypeIdx1, unsigned TypeIdx2,
```
Comment (arsenm) [inline llvm/include/llvm/CodeGen/GlobalISel/LegalizerInfo.h:279]: In the future could consider being fancy and using variadic templates

**Review by arsenm — APPROVED** [review]

File: llvm/lib/Target/AArch64/GISel/AArch64PostLegalizerLowering.cpp [hunk 2]
```diff
@@ -161,6 +161,8 @@ bool matchREV(MachineInstr &MI, MachineRegisterInfo &MRI,
   Register Dst = MI.getOperand(0).getReg();
   Register Src = MI.getOperand(1).getReg();
   LLT Ty = MRI.getType(Dst);
+  if (Ty.isScalableVector())
```
Comment (davemgreen) [inline llvm/lib/Target/AArch64/GISel/AArch64PostLegalizerLowering.cpp:164]: Why are these needed? I don't believe there should be any shuffles on scalable types.

I think that goes for most of the changes in this file. Are the all speculative changes in reality are not necessary?

Comment (tschuett) [reply llvm/lib/Target/AArch64/GISel/AArch64PostLegalizerLowering.cpp:164]: I got asserts and went over almost all of them. The stack trace did not tell me where the crash was.

Comment (tschuett) [reply llvm/lib/Target/AArch64/GISel/AArch64PostLegalizerLowering.cpp:164]: I got asserts and went over almost all of them. The stack trace did not tell me where the crash was.
